pennylane/pauli/grouping/group_observables.py

```diff
@@ -229,8 +230,18 @@ def group_observables(observables, coefficients=None, grouping_type="qwc", metho
     pauli_grouping = PauliGroupingStrategy(
         observables, grouping_type=grouping_type, graph_colourer=method
     )
+
+    temp_opmath = not qml.operation.active_new_opmath() and any(
+        isinstance(o, (Prod, SProd)) for o in observables
+    )
+    if temp_opmath:
+        qml.operation.enable_new_opmath()
```

Comment:
what happens if you don't enable it in this case? (why) do we convert to old-school ops at some point?

Reply:
It's due to [this line](https://github.com/PennyLaneAI/pennylane/blob/d5e3cca6a368a4ef2fccc73dfa6e5ab1f0cbb202/pennylane/pauli/utils.py#L399). Since that function does not have a way of knowing what type of observables were used, it doesn't make sense to update it there.